daliuge-engine/dlg/apps/bash_shell_app.py

```diff
@@ -206,52 +201,31 @@ def _run_bash(self, inputs, outputs, stdin=None, stdout=subprocess.PIPE):
         """
         logger.debug("Parameters found: %s", json.dumps(self.parameters))
         logger.debug("Bash Inputs: %s; Bash Outputs: %s", inputs, outputs)
-        # we only support passing a path for bash apps
-        fsInputs = {uid: i for uid, i in inputs.items() if droputils.has_path(i)}
-        fsOutputs = {uid: o for uid, o in outputs.items() if droputils.has_path(o)}
-        dataURLInputs = {
-            uid: i for uid, i in inputs.items() if not droputils.has_path(i)
-        }
-        dataURLOutputs = {
-            uid: o for uid, o in outputs.items() if not droputils.has_path(o)
-        }
         # deal with named ports
         inport_names = self.parameters["inputs"] if "inputs" in self.parameters else []
         outport_names = (
             self.parameters["outputs"] if "outputs" in self.parameters else []
         )
+
+        cmd = self.command.strip()
+        cmd = droputils.replace_placeholders(cmd, inputs, outputs)
+
         reader = get_port_reader_function(self.input_parser)
         keyargs, pargs = replace_named_ports(
             inputs.items(),
             outputs.items(),
             inport_names,
             outport_names,
             self.appArgs,
-            argumentPrefix=self._argumentPrefix,
-            separator=self._paramValueSeparator,
             parser=reader,
         )
-        argumentString = (
-            f"{' '.join(map(str,pargs + keyargs))}"  # add kwargs to end of pargs
-        )
-        # complete command including all additional parameters and optional redirects
-        if len(argumentString.strip()) > 0:
-            # the _cmdLineArgs would very likely make the command line invalid
-            cmd = f"{self.command} {argumentString} "
-        else:
-            cmd = f"{self.command} {argumentString} {self._cmdLineArgs} "
-        if self._outputRedirect:
-            cmd = f"{cmd} > {self._outputRedirect}"
-        if self._inputRedirect:
-            cmd = f"cat {self._inputRedirect} > {cmd}"
-        cmd = cmd.strip()
-
-        app_uid = self.uid
 
-        # Replace inputs/outputs in command line with paths or data URLs
-        cmd = droputils.replace_path_placeholders(cmd, fsInputs, fsOutputs)
+        for key, value in keyargs.items():
```

Comment:
Set of instructions we discussed to produce replacement:  

```python
>>> of = "my file" 
>>> s = "dd {of}"
>>> eval('f"'+s+'"')
>> 'dd my file'
```

I believe the issue with this approach is we do not have the actual variable names specified; they are keys in the dictionary, so we would have to construct variables dynamically and _then_ perform the `eval`.